### DISCUSSION TITLE
Resolved: Should rustdocs examples use `?`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3033] [General]
It is possible to use `?` in rustdocs but we (I) have favoured using `unwrap`/`expect` because it seemed easier than adding the boilerplate required to use `?`

As we try to make our docs polished should be consider using `?` in rustdocs? 

ref: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#using--in-doc-tests

**Comment by Kixunil** [comment]
FTR it was decided that Rust `std` examples should prefer `?` to teach people proper error handling. I'm in favor of it but I don't consider it critical in this crate. OTOH, I think there's a lot more Rust newbies in BItcoin lately so maybe we should take it into account.

**Reply by storopoli** [reply]
Nice point on bringing what Rust `std` uses. I think I am leaning towards the `?` instead of `unwrap()`/`expect`. I think it brings best practices to our examples. We have plenty of Rust beginner-friendly resources and we should always strive for robustness in our examples.

**Comment by tcharding** [comment]
This is covered by [C-QUESTION-MARK](https://rust-lang.github.io/api-guidelines/documentation.html#c-question-mark). Closing.
